### ISSUE TITLE
[ContainerRegistry] TaskStepProperties's type is incorrectly marked as read-only

### ISSUE THREAD

**Issue by mikhailshilkov** [Azure/azure-rest-api-specs#13891]
The `containerregistry_build` spec (I'm looking at the version 2019-04-01) marks the `type` property of the `TaskStepProperties` schema as read-only: https://github.com/Azure/azure-rest-api-specs/blob/73d51f188ceba91939355c8e1d4d74dd38173c13/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2019-04-01/containerregistry_build.json#L1275

This looks wrong to me because the client is supposed to set this value to "Docker"/"EncodedTask"/"FileTask" depending on which derived type they want to use.

**Comment by guidola** [comment]
Indeed, as the spec is using swagger 2.0 that field needs to be `readOnly: true` , otherwise the polymorphism cannot be properly resolved as the `discriminator` requires it for such purpose.

This is breaking functionality on the endpoint.

**Comment by northtyphoon** [comment]
@mikhailshilkov thanks for reporting the issue. Can you use the following version? The read-only issue has been fixed.

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json#L2196

**Comment by mikhailshilkov** [comment]
Oh, that's good to know! Do you know when the next stable (non-preview) version is going to be published?

**Comment by xyxyxyxyxyxy** [comment]
@mikhailshilkov thank you very much for reporting the issue. It is fixed in our most recent GA version now.
https://github.com/Azure/azure-rest-api-specs/pull/13971
